libs/service-portal/documents/src/hooks/useDocumentList.ts

```diff
@@ -0,0 +1,103 @@
+import { useEffect } from 'react'
+import {
+  AuthDelegationType,
+  DocumentsV2PageOrder,
+  DocumentsV2PageSort,
+} from '@island.is/api/schema'
+import { useUserInfo } from '@island.is/auth/react'
+import { useDocumentContext } from '../screens/Overview/DocumentContext'
+import { useDocumentsV2Query } from '../screens/Overview/Overview.generated'
+import differenceInYears from 'date-fns/differenceInYears'
+
+export const pageSize = 10
+
+export const useDocumentList = () => {
+  const {
+    filterValue,
+    page,
+    totalPages,
+    categoriesAvailable,
+    sendersAvailable,
+
+    setTotalPages,
+    setCategoriesAvailable,
+    setSendersAvailable,
+  } = useDocumentContext()
+
+  const userInfo = useUserInfo()
+  const isLegal = userInfo.profile.delegationType?.includes(
+    AuthDelegationType.LegalGuardian,
+  )
+  const dateOfBirth = userInfo?.profile.dateOfBirth
+  let isOver15 = false
+  if (dateOfBirth) {
+    isOver15 = differenceInYears(new Date(), dateOfBirth) > 15
+  }
+  const hideHealthData = isOver15 && isLegal
+
+  const fetchObject = {
+    input: {
+      senderNationalId: filterValue.activeSenders.join(),
+      nationalId: userInfo.profile.nationalId,
+      dateFrom: filterValue.dateFrom?.toISOString(),
+      dateTo: filterValue.dateTo?.toISOString(),
+      categoryIds: filterValue.activeCategories,
+      subjectContains: filterValue.searchQuery,
+      typeId: null,
+      sortBy: 'Date' as DocumentsV2PageSort,
+      order: 'Descending' as DocumentsV2PageOrder,
```

Comment:
Scary, what if the DocumentsV2PageSort is changed?

Reply:
Ill set these as default values within our graphql. These are basically constants.